File: jambeth/jambeth-ioc-test/src/test/java/de/osthus/ambeth/ioc/performance/IocPerformanceTestModule.java
package de.osthus.ambeth.ioc.performance;

import de.osthus.ambeth.config.Property;
import de.osthus.ambeth.ioc.IInitializingModule;
import de.osthus.ambeth.ioc.factory.IBeanContextFactory;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;

public class IocPerformanceTestModule implements IInitializingModule
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Property(name = IocPerformanceTest.count_prop)
	protected int count;

	@Override
	public void afterPropertiesSet(IBeanContextFactory beanContextFactory) throws Throwable
	{
		for (int a = count; a-- > 0;)
		{
			beanContextFactory.registerBean("name" + a, TestBean.class).propertyValue("Value", "value");
		}
	}
}

File: ambeth/Ambeth.IoC.Test/ambeth/ioc/performance/IocPerformanceTestModule.cs
using De.Osthus.Ambeth.Config;
using De.Osthus.Ambeth.Ioc.Factory;
using De.Osthus.Ambeth.Log;

namespace De.Osthus.Ambeth.Ioc.Performance
{
    public class IocPerformanceTestModule : IInitializingModule
    {
        [LogInstance]
        public ILogger Log { private get; set; }

        [Property(IocPerformanceTest.count_prop)]
        public int Count { protected get; set; }

        public void AfterPropertiesSet(IBeanContextFactory beanContextFactory)
        {
            for (int a = Count; a-- > 0; )
            {
                beanContextFactory.RegisterBean("name" + a, typeof(TestBean)).PropertyValue("Value", "value");
            }
        }
    }
}